Sources/FluentUI_iOS/Components/Pill Button Bar/SwiftUI/PillButtonBarView.swift

```diff
@@ -0,0 +1,235 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+#if canImport(FluentUI_common)
+import FluentUI_common
+#endif
+import SwiftUI
+
+/// `PillButtonBarView` is a horizontal scrollable list of pill shape text buttons.
+/// The bar can either have 1 selected pill at all times or allow no selected pills.
+public struct PillButtonBarView<Selection: Hashable>: View {
+    /// Initializes a new `PillButtonBarView` that allows a the bar to have no selected pill buttons and a maximum of 1 selected pill button.
+    ///
+    /// Use this initializer only if you want to allow the pill button bar to have no selected pills. In other words
+    /// selecting an already selected pill will deselect it.
+    /// - Parameters:
+    ///   - style: The style of the pill buttons.
+    ///   - viewModels: The view model objects representing the pill buttons data.
+    ///   - selected: Optional generic selection binding for the pill bar.
+    ///   - centerAlignIfContentFits: If true, centrally aligns the bar if it fits the container.
+    ///   - tokenOverrides: The token overrides for the pill buttons.
+    public init(style: PillButtonStyle = .primary,
+                viewModels: [PillButtonViewModel<Selection>],
+                selected: Binding<Selection?>,
+                centerAlignIfContentFits: Bool = false,
+                tokenOverrides: [PillButtonToken: ControlTokenValue]? = nil) {
+        self.style = style
+        self.viewModels = viewModels
+        self._selected = selected
+        self.centerAlignIfContentFits = centerAlignIfContentFits
+        self.supportsPillDeselection = true
+        self.tokenOverrides = tokenOverrides
+    }
+
+    /// Initializes a new `PillButtonBarView` that restricts the bar to 1 selected pill button at all times.
+    /// - Parameters:
+    ///   - style: The style of the pill buttons.
+    ///   - viewModels: The view model objects representing the pill buttons data.
```

Comment:
Instead of using view models for the pill buttons have we considered doing something similar to apple's tab view: https://developer.apple.com/documentation/swiftui/tabview

Where we pass in PillButton views with their specifici information (title, image, etc.)